src/components/NavBar.tsx

```diff
@@ -15,7 +17,89 @@ const navItems: NavItem[] = [
   { label: "Ayuda", path: "/ayuda" },
 ];
 
+interface Department {
+  label: string;
+  icon: string;
+}
+
+interface Section {
+  name: string;
+  departments: Department[];
+}
+
+const sections: Section[] = [
+  {
+    name: "Hogar",
+    departments: [
+      { label: "Hogar y muebles", icon: "ri:armchair-line" },
+      { label: "Línea blanca", icon: "icon-park-outline:washing-machine-one" },
+      { label: "Cocina y electrodomésticos", icon: "hugeicons:blender" },
+      { label: "Ferretería y mejoras del hogar", icon: "ion:hammer-outline" },
+    ],
+  },
+  {
+    name: "Tecnología",
+    departments: [
+      { label: "Celulares", icon: "mdi:cellphone" },
+      { label: "Electrónica", icon: "cil:tv" },
+      { label: "Consolas y videojuegos", icon: "solar:gamepad-linear" },
+    ],
+  },
+  {
+    name: "Ropa y Zapatos",
+    departments: [
+      { label: "Zapatos", icon: "hugeicons:running-shoes" },
+      { label: "Bebés", icon: "ic:outline-stroller" },
+      { label: "Niños y Adolescentes", icon: "healthicons:young-people" },
+      { label: "Hombres", icon: "ion:man-outline" },
+      { label: "Mujeres", icon: "ion:woman-outline" },
+    ],
+  },
+  {
+    name: "Accesorios",
+    departments: [
+      {
+        label: "Maletas, Bolsas y Mochilas",
+        icon: "fa6-solid:suitcase-rolling",
+      },
+      { label: "Relojes, Lentes y Joyeria", icon: "mynaui:watch" },
+    ],
+  },
+  {
+    name: "Salud",
+    departments: [
+      { label: "Belleza y Cuidado Personal", icon: "solar:perfume-linear" },
+      { label: "Salud y Bienestar", icon: "solar:hand-heart-outline" },
+    ],
+  },
+  {
+    name: "Movilidad",
+    departments: [
+      { label: "Motos y movilidad", icon: "mdi:racing-helmet" },
+      { label: "Automotriz y Refacciones", icon: "mdi:car-sports" },
+    ],
+  },
+];
+
 const NavBar: React.FC = () => {
+  const [isSubmenuVisible, setIsSubmenuVisible] = useState(false);
+
+  const toggleSubmenu = () => {
+    setIsSubmenuVisible(!isSubmenuVisible);
```

Comment:
No es buena practica mutar el `useState` con su valor, la funcion de `set` del `useState` puede recibir un callback en el cual el parametro de este callback siempre es el estado actual

```suggestion
    setIsSubmenuVisible((prevState) => !prevState);
```